src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs

```diff
@@ -163,28 +168,25 @@ public async Task OnConnectionAsync(ConnectionContext context)
                 await sslStream.AuthenticateAsServerAsync(ServerOptionsCallback, state, cancellationTokenSource.Token);
             }
         }
-        catch (OperationCanceledException)
+        catch (OperationCanceledException ex)
         {
-            KestrelEventSource.Log.TlsHandshakeFailed(context.ConnectionId);
-            KestrelEventSource.Log.TlsHandshakeStop(context, null);
+            RecordHandshakeFailed(_metrics, startTimestamp, Stopwatch.GetTimestamp(), context, ex);
```

Comment:
The exception isn't actually used, just the type name. Pass nameof(OperationCanceledException)? or `RecordHandshakeFailed<OperationCanceledException>(...)`

Reply:
That would lose the name of an exception inherited from `OperationCanceledException`. For example, `TaskCanceledException`.